src/server/frontend_wayland/wl_pointer.cpp

```diff
@@ -113,6 +113,30 @@ auto timestamp_of(MirPointerEvent const* event) -> uint32_t
 {
     return mir_input_event_get_wayland_timestamp(mir_pointer_event_input_event(event));
 }
+
+auto wayland_axis_source(MirPointerAxisSource mir_source) -> std::optional<uint32_t>
+{
+    switch (mir_source)
+    {
+    case mir_pointer_axis_source_none:
+        return std::nullopt;
+
+    case mir_pointer_axis_source_wheel:
+        return static_cast<uint32_t>(mw::Pointer::AxisSource::wheel);
```

Comment:
I have no idea why these `static_cast`s are needed, but both GCC and Clang give
```
undefined symbol: mir::wayland::Pointer::AxisSource::wheel
```
without them.

Reply:
> I have no idea why these `static_cast`s are needed, but both GCC and Clang give

It is because there's no definition of them. Here's the declaration:
```c++
    struct AxisSource
    {
        static uint32_t const wheel = 0;
        static uint32_t const finger = 1;
        static uint32_t const continuous = 2;
        static uint32_t const wheel_tilt = 3;
    };
```

Somewhere there should be a definition:
```C++
uint32_t const mir::wayland::Pointer::AxisSource::wheel;
```
This appears to be a generator problem.